src/DynamoCore/Core/AuthenticationManager.cs

```diff
@@ -33,11 +34,35 @@ public LoginState LoginState
             get { return HasAuthProvider ? authProvider.LoginState : LoginState.LoggedOut; }
         }
 
-        internal bool IsLoggedIn()
+        /// <summary>
+        /// This Property will return the value by checking only once time to the real time LoginState in order to be used in the Initialization flow. Others features require direct calls to the LoginState due to are on demand.
+        /// </summary>
+        public LoginState? LoginStateSingle
         {
+            get
+            {
+                if (singleState == null)
+                {
+                    singleState = LoginState;
+                }
+                return singleState;
+            }
+        }
+
+        internal bool IsLoggedIn()
+        {            
             return HasAuthProvider && authProvider.LoginState == LoginState.LoggedIn ? true : false;
```

Comment:
@jesusalvino Can you comment here if the `authProvider.LoginState` is not performant in low network case?

Reply:
> Can you comment here if the `authProvider.LoginState` is not performant in low network case?

it works but slowly